src/edu/gatech/cs2340/trydent/ContinuousEvent.java

```diff
@@ -0,0 +1,59 @@
+package edu.gatech.cs2340.trydent;
+
+/**
+ * Represents an event that is called by the engine every frame.
+ * @author Garrett Malmquist
+ *
+ */
+public abstract class ContinuousEvent {
```

Comment:
We might represent animations using that, I'm not sure yet. It's also a way for end users to actually hook into the engine in a thread-safe way. They could either just put their game logic in a single ContinuousEvent subclass in update, or subclass a lot of ContinuousEvents to represent different modules of their code (more elegantly, but also require more understanding of the library). This is similar to what Unity does with MonoBehaviors; I think it's the easiest way for users to interact with the engine in a threadsafe way.